draft/ZEP0004.md

```diff
@@ -0,0 +1,194 @@
+
+However, these domains may still benefit from standardization of the way their data and metadata are stored in Zarr.
+For example, they may expect that
+- arrays have specific names
+- specific attributes be present in arrays and groups and conform to certain constraints
+- groups are structured with a specific hierarchy, with defined relationships between different arrays in a group
+
+Such standardization helps interoperability across different languages and packages.
+
+This ZEP aims to formalize the concept of a Zarr **convention**, a mechanism for a domain community to
+document a standard way of storing their specific data model in Zarr.
+As of the time of this draft, many such _ad-hoc_ conventions are already in use across the Zarr landscape.
+A goal of this ZEP is to formalize this existing practice and enhance coordination around conventions.
+
+Conventions must fit completely within the Zarr data / metadata model of groups, arrays, and attributes thereof, requiring
+no changes or extension to the specification.
+A Zarr implementation itself should not even be aware of the existence of the convention.
+The line between a convention and an extension may be blurry in some cases.
+The key distinction lies in the implementation: the responsibility for interpreting a _convention_ relies completely with downstream,
+domain-specific software, while an _extension_ must be handled by the Zarr implementation itself.
+A good rule of thumb is that a user should be able to safely ignore the convention and still be able to interact with the data via the core Zarr library,
+even if some domain-specific context or functionality is missing.
+If the data are completely meaningless or unintelligible without the convention, then it should be an extension instead.
+
+It seems likely that some features that are first supported as conventions may be promoted to extensions if they are deemed sufficiently general.
+
+Conventions can also help users switch between different storage libraries more flexibly.
+Since Zarr and HDF5 implement nearly identical data models, a single convention can be applied to both formats.
+This allows downstream software to maintain better separation of concerns between storage and domain-specific logic.
+
+## Usage and Impact
+
+We demonstrate the usage and impact with a simple example: unit encoding.
+
+A convention around unit encoding might look like this
+
+> **Convention Name**: "units-v1"
+>
+> Information about the units of a Zarr array should be stored in the `units` attribute and encoded as strings using
+> the [MIXF-08](http://people.csail.mit.edu/jaffer/MIXF/MIXF-08) standard.
+
+To create data with this convention using zarr-python, one would write
+
+```python
+import zarr
+z = zarr.open(
+    'example_with_units.zarr', mode='w', shape=(10000, 10000), chunks=(1000, 1000), dtype='f4'
+)
+z.attrs['units'] = 'm^2' 
+z.attrs['zarr_convention'] = "units-v1"
```

Comment:
For me, this is the most (or really only) potentially contentious line in the ZEP. Similar to the ZEP1 discussion last night, I could see discussing at least:

 - if `zarr_` is a generally special prefix or if `zarr_convention` is a special string
 - if it's possible to have more than one convention active in a zgroup/zarray at a time
 - whether other values (like URLs) are acceptable as the values of the convention string

Reply:
I'm also wondering if this is needed at all. Instead, I'd rather just specify that the field `units` has a metadata convention. Nobody is forced to follow this convention, and exisiting arrays with such a field would potentially follow it automatically.